### PR TITLE
engine selection at service creation time : CAFFE of CUDNN

### PR DESCRIPTION
**Pull request by fantes**
this PR allows to select engine (cudnn or caffe (which is CPU or GPU depending on GPU availability)) at service creation time, for layers that can, ie conv, deconv, tanh, relu, sigmoid, pooling, lrn, and softmax.

 This is a workaround to memory over consumption / leakage seen with some models that do a _lot_ of cudnncreate 

if unset, use default, which is cudnn is deepdetect+caffe was compiled with -DUSE_CUDNN=ON , or caffe in other case

parameter is parameters.mllib.engine  , can be set to "CUDNN" or "CAFFE"

example: 
curl -X PUT http://localhost:8080/services/ocr -d '{
 "description": "ocr",
 "model": {
  "repository": "/home/infantes/airbusey_comp_ocr_resnext50_v4"
 },
 "mllib": "caffe",
 "type": "supervised",
 "parameters": {
  "input": {
   "connector": "image",
   "width":250,
   "height":200, "ctc":true, "bw":false, "db":true
  },
  "mllib" : {
     "nclasses":40, "finetuning": true,
     "rotate":false,
     "mirror":false,
     "scale":1,
     "engine": "CUDNN",
     "db":true,
     "noise": {
          "all_effects":true,
          "prob":0.01
     },
    "distort": {
        "all_effects":true,
        "prob":0.01
    },
    "geometry" : {
         "persp_horizontal":true,
         "persp_vertical":true,
         "zoom_out":true,
         "pad_mode":"constant",
         "prob":0.001
     },
     "gpu":true,
     "gpuid":1

  }
 }
}'

### CODE REVIEW

**Comment by beniz** [comment]
Maybe let's rebase on master.
